docs/java-rest/high-level/security/authenticate.asciidoc

```diff
@@ -0,0 +1,56 @@
+[[java-rest-high-security-authenticate]]
+=== Authenticate API
+
+[[java-rest-high-security-authenticate-execution]]
+==== Execution
+
+To authenticate and retrieve information about a user can be performed
+using the `security().authenticate()` method:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SecurityDocumentationIT.java[authenticate-execute]
+--------------------------------------------------
+
+This method does not require a request object.
+
+[[java-rest-high-security-authenticate-response]]
+==== Response
+
+The returned `AuthenticateResponse` contains a single field, `user`. This field
+, accessed with `getUser`, contains all the information about this
+authenticated user.
```

Comment:
I wonder if could link to javadocs here ? I would like to link to thee User javadoc class.

Reply:
I'm not sure about that, maybe @nik9000 knows? I am pretty sure he was working on some improvements around the docs for the HLRC

Reply:
there is a way to point to the root Javadoc, see `/docs/java-rest/high-level/getting-started.asciidoc` but I wanted a link to a specific class's javadoc. There does not seem to be any practical way for this (other than crafting rickety links).

Reply:
@albertzaharovits We can make the links slightly-less rickety by adding shared attributes for the javadoc package URLs. For example, if we define:

:javadoc-client:       {rest-high-level-client-javadoc}/org/elasticsearch/client

Then you can link to specific classes in the client package by specifying {javadoc-client}/<classname>.html. 

If that seems workable, I can add the attributes to the Versions.asciidoc file.

Reply:
@debadair 
I will be trying it, thank you for the air support!